### PR TITLE
Enhance TFTrainer.save_model()

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

@jplu , could you close PR #7597, please? This PR is a clean one, without the file `modeling_tf_utils` being changed at all.

Currently, TFTrainer.save_model() raises errors if the model is not TFPreTrainedModel . However Trainer works fine with torch.nn.modules.Module.

This is a step to make TFTrainer work with usual tf.keras.models.Model models. The idea (from @sgugger) is that a user is building their own models that work like ours (e.g., return the loss as the first output) and can train them with Trainer.

Furthermore, a SavedModel is also saved using tf.saved_model.save().

For @jplu and @sgugger .

### CODE REVIEW

**Review by jplu — APPROVED** [review]
Awesome! Did you try it with a usual training example to see if everything is ok?

**Comment by chiapas** [comment]
> 
> 
> Awesome! Did you try it with a usual training example to see if everything is ok?

Yes, with example/text-classification. I didn't really look at the accuracy though (computer too slow). But it works fine.
I didn't check yet with a usual `tf.keras.models.Model` (i.e. not TFPretrainedModel). But when I continue with `test_trainer_tf.py`, it will be tested.

**Review by LysandreJik — APPROVED** [review]
LGTM!

File: src/transformers/trainer_tf.py [hunk 1]
```diff
@@ -785,6 +786,22 @@ def save_model(self, output_dir: Optional[str] = None):
         logger.info("Saving model in {}".format(output_dir))
 
         if not isinstance(self.model, TFPreTrainedModel):
-            raise ValueError("Trainer.model appears to not be a PreTrainedModel")
+            warnings.warn(
+                "Trainer.model appears to not be a PreTrainedModel. The model will still be saved as a usual tf.keras.models.Model model, but no PretrainedConfig will be saved."
```
Comment (sgugger) [inline src/transformers/trainer_tf.py:790]:
```suggestion
                "`Trainer.model` appears to not be a `PreTrainedModel`. The model will still be saved as a usual `tf.keras.models.Model`, but no `PretrainedConfig` will be saved."
```

**Review by sgugger — APPROVED** [review]
Just a nit, but LGTM too, thanks for your work!

**Comment by ydshieh** [comment]
Hi, @sgugger & @LysandreJik ,

I didn't realized that this PR is not merged into master. Since it has been for some time, I rebased the branch. All the suggestions from @sgugger  are done in the latest version.

It would be great if you can merge  this PR. Thanks!
